cdap-ui/app/cdap/components/DataPrep/TopPanel/SchemaModal.js

```diff
@@ -70,7 +70,15 @@ export default class SchemaModal extends Component {
           fields: res
         };
 
-        let fields = getParsedSchema(tempSchema);
```

Comment:
Instead of creating a new function, couldn't we just check if the return result of `getParsedSchema` is undefined, then set the error state if it is?